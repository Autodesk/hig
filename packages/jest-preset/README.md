# @hig/jest-preset

Jest preset for HIG packages.

## Usage

_Example `package.json`:_

```json
{
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
```
