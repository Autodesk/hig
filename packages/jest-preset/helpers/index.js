/* eslint-disable global-require */
module.exports = {
  takeSnapshotsOf: require("./takeSnapshotsOf")
};
