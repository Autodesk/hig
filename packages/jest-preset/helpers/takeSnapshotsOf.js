const React = require("react");
const renderer = require("react-test-renderer");

module.exports = function takeSnapshotsOf(Component, cases, rendererOptions) {
  cases.forEach(({ desc, props }) => {
    it(desc, () => {
      const vDom = React.createElement(Component, props);
      const tree = renderer.create(vDom, rendererOptions).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
};
