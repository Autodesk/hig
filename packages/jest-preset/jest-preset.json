{
  "collectCoverageFrom": ["**/src/**/*.{js,jsx}"],
  "coverageDirectory": "<rootDir>/coverage",
  "coveragePathIgnorePatterns": [
    "__gemini__",
    "__stories__",
    "docs",
    "examples/redux",
    "node_modules",
    "packages/jest-preset",
    "packages/vanilla",
    "src/playground"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 50,
      "functions": 50,
      "lines": 50,
      "statements": 50
    }
  },
  "moduleNameMapper": {
    "\\.(css|scss|svg)$": "@hig/jest-preset/fileMock.js"
  },
  "modulePaths": ["node_modules"],
  "moduleFileExtensions": ["js", "jsx", "json"],
  "setupFiles": ["raf/polyfill"],
  "setupTestFrameworkScriptFile": "@hig/jest-preset/setupTests.js",
  "testPathIgnorePatterns": [
    "docs",
    "packages/babel-preset",
    "packages/jest-preset",
    "packages/vanilla"
  ],
  "unmockedModulePathPatterns": ["node_modules"]
}
