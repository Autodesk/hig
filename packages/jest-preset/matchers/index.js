require("./toBeConstants");
require("./toImplementHIGInterfaceOf");
require("./toHavePropTypesAndDocgenInfo");
