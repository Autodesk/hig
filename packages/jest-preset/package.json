{
  "private": true,
  "name": "@hig/jest-preset",
  "version": "0.1.0",
  "devDependencies": {
    "babel-core": "^6.26.3",
    "babel-jest": "^23.4.0",
    "enzyme": "^2.9.1",
    "jest": "^21.2.1",
    "jest-enzyme": "^3.8.3",
    "jest-extended": "^0.6.0",
    "raf": "^3.4.0",
    "react-test-renderer": "^15.6.1"
  },
  "eslintConfig": {
    "extends": "@hig"
  },
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
