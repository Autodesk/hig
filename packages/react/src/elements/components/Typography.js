import React from "react";
import PropTypes from "prop-types";
import Typography from "../../adapters/TypographyAdapter";

function H1Component(props) {
  return <Typography type="h1" text={props.children} />;
}

function H2Component(props) {
  return <Typography type="h2" text={props.children} />;
}

function H3Component(props) {
  return <Typography type="h3" text={props.children} />;
}

function Sub1Component(props) {
  return <Typography type="sub1" text={props.children} />;
}

function Sub2Component(props) {
  return <Typography type="sub2" text={props.children} />;
}

function BodyComponent(props) {
  return <Typography type="body" text={props.children} />;
}

function BoldComponent(props) {
  return <Typography type="bold" text={props.children} />;
}

function DisabledComponent(props) {
  return <Typography type="disabled" text={props.children} />;
}

function CaptionComponent(props) {
  return <Typography type="caption" text={props.children} />;
}

const docgenInfo = {
  props: {
    children: {
      description: "any content"
    }
  }
};

const propTypes = {
  children: PropTypes.node
};

[
  H1Component,
  H2Component,
  H3Component,
  Sub1Component,
  Sub2Component,
  BodyComponent,
  BoldComponent,
  DisabledComponent,
  CaptionComponent
].forEach(componentClass => {
  componentClass.__docgenInfo = docgenInfo; // eslint-disable-line no-param-reassign
  componentClass.propTypes = propTypes; // eslint-disable-line no-param-reassign
});

export {
  H1Component as H1,
  H2Component as H2,
  H3Component as H3,
  Sub1Component as Sub1,
  Sub2Component as Sub2,
  BodyComponent as Body,
  BoldComponent as Bold,
  DisabledComponent as Disabled,
  CaptionComponent as Caption
};
