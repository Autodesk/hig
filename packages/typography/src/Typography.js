import React, { Component } from "react";
import PropTypes from "prop-types";
import { css } from "emotion";
import ThemeContext from "@hig/theme-context";

import {
  AVAILABLE_ALIGNMENTS,
  AVAILABLE_FONT_WEIGHTS,
  AVAILABLE_VARIANTS
} from "./_constants";

import stylesheet from "./Typography.stylesheet";

export default class Typography extends Component {
  static propTypes = {
    /**
     * Sets the horizontal alignment of the text
     */
    align: PropTypes.oneOf(AVAILABLE_ALIGNMENTS),
    /**
     * Enables specifying the semantic element to be rendered by the component
     * If this prop is not provided, the semantic element will default to matching the variant (e.g. if the component
     * variant is h1, the element will be `<h1>`) or if there is no semantic element matching the variant, it
     * will default to `<p>`. You can provide elementType as a string, like "figcaption", or as a function, like
     * `({children}) => (<figcaption>{children}</figcaption>)`.
     */
    elementType: PropTypes.node,
    /**
     * Text to render
     */
    children: PropTypes.node,
    /**
     * Specifies the weight (or boldness) of the font
     */
    fontWeight: PropTypes.oneOf(AVAILABLE_FONT_WEIGHTS),
    /**
     * Indicates the initial Typography style
     */
    variant: PropTypes.oneOf(AVAILABLE_VARIANTS)
  };

  elementType = () => {
    const { elementType, variant } = this.props;

    if (elementType) {
      return elementType;
    }

    return ["h1", "h2", "h3"].includes(variant) ? variant : "p";
  };

  render() {
    const {
      align,
      children,
      fontWeight,
      variant,
      elementType, // we don't want this included in the otherProps that appear in the DOM
      ...otherProps
    } = this.props;

    return (
      <ThemeContext.Consumer>
        {({ resolvedRoles }) => {
          const styles = stylesheet(
            { align, fontWeight, variant },
            resolvedRoles
          );

          const ElementType = this.elementType();

          return (
            <ElementType className={css(styles.typography)} {...otherProps}>
              {children}
            </ElementType>
          );
        }}
      </ThemeContext.Consumer>
    );
  }
}
