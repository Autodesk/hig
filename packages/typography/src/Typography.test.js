import React from "react";
import { shallow } from "enzyme";
import { takeSnapshotsOf } from "@hig/jest-preset/helpers";
import Typography from "./Typography";

describe("Typography", () => {
  function props({ ...args }) {
    const { align, elementType, fontWeight, variant } = args;
    const variantDetails = variant ? `${variant} variant` : "Default";
    const otherDetails =
      !!align || !!fontWeight || !!elementType
        ? ` with ${[align, fontWeight, elementType].join(" ")}`
        : "";

    return {
      ...args,
      children: `${variantDetails} should render nicely${otherDetails}.`
    };
  }

  it("renders", () => {
    expect(
      shallow(<Typography>It should render nicely</Typography>)
    ).toMatchObject(/It should render nicely/);
  });

  takeSnapshotsOf(Typography, [
    // new API tests
    { description: "renders default Typography", props: props() },
    {
      description: "renders Typography with align and fontWeight props",
      props: props({ align: "center", fontWeight: "bold", variant: "body" })
    },
    {
      description: "renders body variant Typography",
      props: props({ variant: "body" })
    },
    {
      description: "renders caption variant Typography",
      props: props({ variant: "caption" })
    },
    {
      description: "renders h1 variant Typography",
      props: props({ variant: "h1" })
    },
    {
      description: "renders h2 variant Typography",
      props: props({ variant: "h2" })
    },
    {
      description: "renders h3 variant Typography",
      props: props({ variant: "h3" })
    },
    {
      description:
        "renders h3 variant Typography with any arbitrary semantic element",
      props: props({ variant: "h3", elementType: "figcaption" })
    }
  ]);
});
