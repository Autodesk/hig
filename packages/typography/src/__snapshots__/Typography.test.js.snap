// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography  1`] = `
.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.3;
  margin: 0;
  text-align: initial;
}

<p
  className="emotion-0"
>
  Default should render nicely.
</p>
`;

exports[`Typography  2`] = `
.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: bold;
  line-height: 1.3;
  margin: 0;
  text-align: center;
}

<p
  className="emotion-0"
>
  body variant should render nicely with center bold .
</p>
`;

exports[`Typography  3`] = `
.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.3;
  margin: 0;
  text-align: initial;
}

<p
  className="emotion-0"
>
  body variant should render nicely.
</p>
`;

exports[`Typography  4`] = `
.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 12px;
  font-weight: 400;
  line-height: 1.3;
  margin: 0;
  text-align: initial;
}

<p
  className="emotion-0"
>
  caption variant should render nicely.
</p>
`;

exports[`Typography  5`] = `
.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 28px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  text-align: initial;
}

<h1
  className="emotion-0"
>
  h1 variant should render nicely.
</h1>
`;

exports[`Typography  6`] = `
.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 24px;
  font-weight: 400;
  line-height: 1.3;
  margin: 0;
  text-align: initial;
}

<h2
  className="emotion-0"
>
  h2 variant should render nicely.
</h2>
`;

exports[`Typography  7`] = `
.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 20px;
  font-weight: 600;
  line-height: 1.4;
  margin: 0;
  text-align: initial;
}

<h3
  className="emotion-0"
>
  h3 variant should render nicely.
</h3>
`;

exports[`Typography  8`] = `
.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 20px;
  font-weight: 600;
  line-height: 1.4;
  margin: 0;
  text-align: initial;
}

<figcaption
  className="emotion-0"
>
  h3 variant should render nicely with   figcaption.
</figcaption>
`;
